lib/btrfs_test.pm

```diff
@@ -18,13 +19,49 @@ sub set_unpartitioned_disk_in_bash {
         $vd = 'sd';
     }
     assert_script_run 'parted --script --machine -l';
-    assert_script_run 'disk=${disk:-$(parted --script --machine -l |& sed -n \'s@^\(/dev/' . $vd . '[ab]\):.*unknown.*$@\1@p\')}';
+    assert_script_run 'disk=${disk:-$(parted --script --machine -l |& sed -n \'s@^\(/dev/' . $vd . 'b\):.*$@\1@p\')}';
     assert_script_run 'echo $disk';
 }
 
 sub cleanup_partition_table {
     assert_script_run 'wipefs --force --all $disk';
 }
 
+=head2 snapper_nodbus_setup
+
+In `snapper --no-dbus` test we need DBus to be disabled on SLES12SP3 and Leap 42.3
+systemd allows DBus to be disabled. On Tumbleweed this is not possible and the simplest
+way to get DBus-less environment is to enter rescue.target via systemctl.
+=cut
+sub snapper_nodbus_setup {
+    my ($self) = @_;
+    if (script_run('! systemctl is-active dbus')) {
+        script_run('systemctl rescue', 0);
+        assert_screen('emergency-shell', 10);
+        type_password;
+        send_key 'ret';
+        $self->set_standard_prompt('root');
+        assert_screen 'root-console';
+    }
```

Comment:
in snapper_nodbus_restore we dying in case dbus is not in expected state. Here it is not so critical ?